File: PYTHON_SERVER/app/model/ProductAttributeValueSummary.py
from sqlalchemy.orm import relationship

from app.database import Base
from sqlalchemy import Column, Integer, String, DateTime, ForeignKey
from sqlalchemy.sql import func


class ProductAttributeValueSummary(Base):
    __tablename__ = 'product_attribute_value_summary'

    id = Column(Integer, primary_key=True, autoincrement=True)
    job_id = Column(Integer, index=True)  # 🔥 Job ID
    product_id = Column(Integer, index=True)  # 🔥 Product ID
    company_id = Column(Integer, ForeignKey('company.id'))
    attribute_id = Column(Integer, ForeignKey('attribute.id'))
    mpn = Column(String, index=True)
    value = Column(String)
    created_at = Column(DateTime(timezone=True), server_default=func.now())
    updated_at = Column(DateTime(timezone=True), server_default=func.now(), onupdate=func.now())

    company = relationship("Company", back_populates="product_attribute_value_summary")
    attribute = relationship("Attribute", back_populates="product_attribute_value_summary")

File: PANEL/app/Models/ProductAttributeValueSummary.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class ProductAttributeValueSummary extends Model
{
    use HasFactory;

    protected $table = 'product_attribute_value_summary';

    protected $fillable = [
        'company_id',
        'value',
        'job_id',
        'product_id',
        'attribute_id',
        'mpn',
    ];

    /**
     * Get the company that owns the product attribute value summary.
     */
    public function company()
    {
        return $this->belongsTo(Company::class);
    }

    /**
     * Get the attribute that owns the product attribute value summary.
     */
    public function attribute()
    {
        return $this->belongsTo(Attribute::class);
    }
}

